2025-03-09 22:13:51,286 INFO    Thread-16 (_run_job):16656 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221351-ocjrl3jb\logs\debug.log
2025-03-09 22:13:51,286 INFO    Thread-16 (_run_job):16656 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221351-ocjrl3jb\logs\debug-internal.log
2025-03-09 22:13:51,286 INFO    Thread-16 (_run_job):16656 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:13:51,286 INFO    Thread-16 (_run_job):16656 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'relu', 'batchSize': 164, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 5, 'num_hiddenLayers': 4, 'optimizer': 'momentum', 'weightDecay': 0, 'weightInit': 'xavier'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:13:51,286 INFO    Thread-16 (_run_job):16656 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:13:51,419 INFO    Thread-16 (_run_job):16656 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/ocjrl3jb
2025-03-09 22:13:51,420 ERROR   Thread-16 (_run_job):16656 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:13:51,420 INFO    Thread-16 (_run_job):16656 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:13:51,420 INFO    Thread-16 (_run_job):16656 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:13:51,420 INFO    Thread-16 (_run_job):16656 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:13:51,420 INFO    Thread-16 (_run_job):16656 [wandb_run.py:_restore():2322] restore
2025-03-09 22:13:51,420 INFO    Thread-16 (_run_job):16656 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:13:53,471 INFO    Thread-16 (_run_job):16656 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:13:53,471 INFO    Thread-16 (_run_job):16656 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:13:53,471 INFO    Thread-16 (_run_job):16656 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:13:53,476 ERROR   MainThread:16656 [pyagent.py:_run_jobs_from_queue():234] Run ocjrl3jb errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_16656\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_16656\3783146247.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_16656\3783146247.py", line 129, in backpropagation
    deltas = [self.layers[-1] - y]  # Output layer error
              ~~~~~~~~~~~~~~~~^~~
ValueError: operands could not be broadcast together with shapes (164,10) (10,164) 

2025-03-09 22:14:03,523 INFO    MsgRouterThr:16656 [mailbox.py:close():129] Closing mailbox, abandoning 1 handles.
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_setup.py:_flush():67] Configure stats pid to 16656
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_setup.py:_flush():67] Loading settings from C:\Users\prajv\.config\wandb\settings
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_setup.py:_flush():67] Loading settings from c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\settings
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221404-zpd2idxq\logs\debug.log
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221404-zpd2idxq\logs\debug-internal.log
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_init.py:monkeypatch_ipython():599] configuring jupyter hooks <wandb.sdk.wandb_init._WandbInit object at 0x000002AB674E0E00>
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 116, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 5, 'num_hiddenLayers': 4, 'optimizer': 'momentum', 'weightDecay': 0.0005, 'weightInit': 'xavier'}
config: {'_wandb': {}}
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():784] starting backend
2025-03-09 22:14:04,553 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():788] sending inform_init request
2025-03-09 22:14:04,558 INFO    Thread-21 (_run_job):16656 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=spawn, using: spawn
2025-03-09 22:14:04,558 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():798] backend started and connected
2025-03-09 22:14:04,560 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_config_callback():1261] config_cb None None {'activationFunc': 'tanh', 'batchSize': 116, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 5, 'num_hiddenLayers': 4, 'optimizer': 'momentum', 'weightDecay': 0.0005, 'weightInit': 'xavier'}
2025-03-09 22:14:04,563 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_label_probe_notebook():1204] probe notebook
2025-03-09 22:14:04,583 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():891] updated telemetry
2025-03-09 22:14:04,664 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-09 22:14:06,496 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():990] starting run threads in backend
2025-03-09 22:14:06,826 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_console_start():2375] atexit reg
2025-03-09 22:14:06,826 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-09 22:14:06,826 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-09 22:14:06,827 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-09 22:14:06,828 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-09 22:14:06,831 INFO    Thread-21 (_run_job):16656 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221406-zpd2idxq\logs\debug.log
2025-03-09 22:14:06,834 INFO    Thread-21 (_run_job):16656 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221406-zpd2idxq\logs\debug-internal.log
2025-03-09 22:14:06,834 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:14:06,834 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 116, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 5, 'num_hiddenLayers': 4, 'optimizer': 'momentum', 'weightDecay': 0.0005, 'weightInit': 'xavier'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:14:06,834 INFO    Thread-21 (_run_job):16656 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:14:06,984 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/zpd2idxq
2025-03-09 22:14:06,986 ERROR   Thread-21 (_run_job):16656 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:14:06,986 INFO    Thread-21 (_run_job):16656 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:14:06,986 INFO    Thread-21 (_run_job):16656 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:14:06,986 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:14:06,986 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_restore():2322] restore
2025-03-09 22:14:06,986 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:14:08,927 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:14:08,927 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:14:08,927 INFO    Thread-21 (_run_job):16656 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:14:08,935 ERROR   MainThread:16656 [pyagent.py:_run_jobs_from_queue():234] Run zpd2idxq errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_16656\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_16656\3783146247.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_16656\3783146247.py", line 129, in backpropagation
    deltas = [self.layers[-1] - y]  # Output layer error
              ~~~~~~~~~~~~~~~~^~~
ValueError: operands could not be broadcast together with shapes (116,10) (10,116) 

2025-03-09 22:14:08,939 ERROR   MainThread:16656 [pyagent.py:_run_jobs_from_queue():245] Detected 3 failed runs in the first 60 seconds, killing sweep.
